### PR TITLE
Add Technical Documentation for Webchat Usage

### PR DESCRIPTION
**Pull request by pedily**

### CODE REVIEW

File: docs/embedding.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+| persistentMenu | [Persitent Menu](#persistent-menu) | - | The Persistent Menu to render in the Webchat.
+
+See it in action:  
+[![Edit Override Endpoint Settings](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/embedding-the-cognigy-webchat-bpz1r?fontsize=14&hidenavigation=1&theme=dark)
+
+#### Persistent Menu
+| Name | Type | Default | Description |
+| - | - | - | - |
+| title | string | `""` | The title of the Persistent Menu |
+| menuItems | List of [Persitent Menu Item](#persistent-menu-item)s | `[]` | A List of Items that should appear in the Persistent Menu
```
Comment (dshire) [inline docs/embedding.md]: Typo "Persitent". And better move the s into the brackets, else it can look a bit silly

File: docs/embedding.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+| colorScheme | string | The background color of the header and bot messages in the Webchat. | designTemplate | 1 or 2 | 1 | The Webchat design template to use. We default to design template 1 (bottom right with a button), you can switch to template 2, which is the centered webchat. 
+| messageLogoUrl | string | COGNIGY.AI Logo | The logo to display next to the messages from your bot in the Webchat. Defaults to a COGNIGY.AI logo. 
+| headerLogoUrl | string | COGNIGY.AI Logo | The logo to display in the header of the Webchat. Defaults to a COGNIGY.AI logo. 
+| enableConnectionStatusIndicator | boolean | false | Whether to show a warning if the connection is lost during a conversation. The warning will disappear when the connection is re-established. | enableTypingIndicator | boolean | true | Whether to enable typing indicators in the Webchat when the Conversational AI is replying. Requires a messageDelay to be set. 
+| enablePersistentMenu | boolean | false | Whether to enable the Persistent Menu 
+| persistentMenu | [Persitent Menu](#persistent-menu) | - | The Persistent Menu to render in the Webchat.
```
Comment (dshire) [inline docs/embedding.md]: Typo "Persitent"

File: docs/embedding.md (hunk 1 above)
Comment (dshire) [inline docs/embedding.md]: Type field should read "Array of ..."

File: docs/embedding.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+| Name | Type | Default | Description |
+| - | - | - | - |
+| title | string | `""` | The title of the Persistent Menu |
+| menuItems | List of [Persitent Menu Item](#persistent-menu-item)s | `[]` | A List of Items that should appear in the Persistent Menu
+
+#### Persistent Menu Item
+| Name | Type | Default | Description |
+| - | - | - | - |
+| title | string | `""` | The label of the Persisted Menu Item and visible Text on the Message |
+| payload | string | inherited from `title` | The actual text message that should be sent |
```
Comment (dshire) [inline docs/embedding.md]: Possible Bug: In my test, the inheritance from title didn't work. I had no response when clicking a menu item that had only the title defined, even though the input was displayed in the chat window

Comment (dshire) [reply docs/embedding.md]: [example here](https://codesandbox.io/s/override-endpoint-settings-kkeq6?fontsize=14&hidenavigation=1&theme=dark)

File: docs/analytics-api.md [hunk 4]
```diff
@@ -1,78 +1,36 @@
...
+| `plugin/text-input/open-persistent-menu` | - | The persistent menu was opened |
+| `plugin/text-input/close-persistent-menu` | - | The persistent menu was closed |
+| `plugin/text-input/click-persistent-menu-item` | `{ text }` | A persistent menu item was clicked |
+| `plugin/messenger/action` | `Object` | An action was triggered from a Webchat or Messenger Template | 
+
+If you e.g. want to react to incoming messages from Cognigy, you can do it like this:
+```javascript
+webchat.registerAnalyticsService(event => {
+    if (event.type === "webchat/incoming-messsage") {
+        console.log("incoming message: " + event.payload.text);
```
[the remark this answers is not in the document]
Comment (pedily) [reply docs/analytics-api.md]: fixed

File: docs/css-customization.md [hunk 5]
```diff
@@ -375,171 +375,5 @@ This is the button to open the Webchat, if you want to modify its color you have
 
 Here is an example that you could put in your code directly and see how it changes the whole Webchat. We have left some of the properties empty for the sake of the design, but we let them there so you can get the whole picture!
```
[the remark this answers is not in the document]
Comment (pedily) [reply docs/css-customization.md]: i removed this part

File: docs/css-customization.md [hunk 6]
```diff
@@ -375,171 +375,5 @@ This is the button to open the Webchat, if you want to modify its color you have
...
-    }
-    
-    [data-cognigy-webchat-root] [data-cognigy-webchat-toggle].webchat-toggle-button {
-    
-        background-image: none;
-        background-color: rgb(88, 195, 229);
-    }
-    
-</style>
+See it in action:
```
[the remark this answers is not in the document]
Comment (pedily) [reply docs/css-customization.md]: i removed the text above and moved the example to the top

File: docs/embedding.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+## Basic Implementation
+To integrate the Webchat into a Website, you need to
+1. load the `webchat.js` bundle via a `<script>` tag
+2. initialize the Webchat towards a Cognigy Endpoint using `initWebchat()`
+
+See it in action:  
+[![Edit Basic Implementation](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/basic-cognigy-webchat-embedding-ict1u?fontsize=14&hidenavigation=1&theme=dark)
+
+## Using a Compatiblity Build
+For older browsers, we ship a seperate build of the Webchat called `webchat.legacy.js`, which comes with extra compatibility at the cost of increased size.
```
[the remark this answers is not in the document]
Comment (pedily) [reply docs/embedding.md]: i copied the infos from our readme docs here

File: docs/embedding.md (hunk 1 above)
Comment (pedily) [reply docs/embedding.md]: fixed, and fixed.

File: docs/embedding.md (hunk 3 above)
Comment (pedily) [reply docs/embedding.md]: fixed

**Pull request merged**